src/btree/bt_sync_obsolete.c

```diff
@@ -228,29 +255,14 @@ __sync_obsolete_cleanup_one(WT_SESSION_IMPL *session, WT_REF *ref)
          */
         new_state = previous_state;
         if (previous_state == WT_REF_DELETED)
-            ret = __sync_obsolete_deleted_cleanup(session, ref);
+            ret = __sync_obsolete_deleted_cleanup(session, ref, num_cc_pagesp);
         else if (previous_state == WT_REF_DISK) {
-            ret = __sync_obsolete_disk_cleanup(session, ref, &ref_deleted);
+            ret = __sync_obsolete_disk_cleanup(session, ref, &ref_deleted, num_cc_pagesp);
             if (ref_deleted)
                 new_state = WT_REF_DELETED;
         }
         WT_REF_UNLOCK(ref, new_state);
         WT_RET(ret);
-    } else if (ref_state == WT_REF_MEM) {
```

Comment:
Why this is removed?

Reply:
This scenario is handled in another code flow.